### PR TITLE
feat(table): added fixed columns

### LINKED ISSUE #4218: Table scrolling and sticky column
As a user, I want to horizontally scroll through a table with daily data. 

In Cost Management, we're implementing a new "Historical Cost Explorer" feature. This charts multiple months worth of daily data. However, we also have the requirement to show a more detailed view of the data in a table (below). To view daily data in a table, we need the ability to scroll horizontally.

We could place the entire table in a scrollable window; however, the Drift app (another Insights application), has created their own custom table to support scrolling. In the screenshot below, the first column is sticky, while the other columns are allowed to scroll horizontally. This is the basic functionality we need from the PatternFly table.

**Drift**
chrome-capture

**Historical Cost Explorer**
Screen Shot 2021-01-21 at 11 32 18 AM

### PR DESCRIPTION
**Pull request by mattnolting**
closes #4218 

Design: https://marvelapp.com/prototype/5d421g0

### CODE REVIEW

File: src/patternfly/components/Table/table-scrollable.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
+.pf-c-table {
+  --pf-c-table__sticky-column--cell-min-width--base: 200px;
+  --pf-c-table__sticky-column--MinWidth: var(--pf-c-table__sticky-column--cell-min-width--base);
+  --pf-c-table__sticky-column--Left: 0;
+  --pf-c-table__sticky-column--ZIndex: var(--pf-global--ZIndex--sm);
+  --pf-c-table__sticky-column--Top: 0;
+  --pf-c-table__sticky-column--BackgroundColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-table__sticky-column--before--BorderRightWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-table__sticky-column--before--BorderRightColor: var(--pf-global--BorderColor--100);
+  --pf-c-table--thead__sticky-column--ZIndex: calc(var(--pf-global--ZIndex--sm) + 1);
```
Comment (mcoker) [inline src/patternfly/components/Table/table-scrollable.scss:10]: doesn't look like it's used

**Review by mcoker — CHANGES_REQUESTED** [review]
Amazing! I think the approach is solid. Left some feedback. Also the borders in FF look a little wonky.

Screen Shot 2021-09-07 at 11 01 54 AM

File: src/patternfly/components/Table/templates/table--scrollable--tbody--tr.hbs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{{#> table-tr}}
+  {{#if table--scrollable--Column1IsStickyColumn}}
+    {{> table--scrollable--th table--scrollable--th--modifier=table--scrollable--th--modifier--cell-1-modifier table--scrollable--th--content=cell-1--content table-th--IsStickyColumn="true" table--scrollable--NoIcon="true"}}
+  {{else}}
+    {{> table--scrollable--td table--scrollable--td--content=cell-1--content}}
+  {{/if}}
+  {{#if table--scrollable--Column2IsStickyColumn}}
+    {{> table--scrollable--th table--scrollable--th--modifier=table--scrollable--th--modifier--cell-2-modifier table--scrollable--th--content=cell-2--content table-th--IsStickyColumn="true" table-th--attribute='style="--pf-c-table__sticky-column--MinWidth: 200px; --pf-c-table__sticky-column--Left: 200px;"'}}
```
Comment (mcoker) [inline src/patternfly/components/Table/templates/table--scrollable--tbody--tr.hbs:8]: `--pf-c-table__sticky-column--MinWidth: 200px` is the default. Do you want to use that here? Or do you want to set that to some other value on a column to show how it changes the left value of adjacent columns?

**Review by mcoker — CHANGES_REQUESTED** [review]

File: src/patternfly/components/Table/table-scrollable.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
+.pf-c-table {
+  --pf-c-table__sticky-column--cell-min-width--base: 200px;
+  --pf-c-table__sticky-column--MinWidth: var(--pf-c-table__sticky-column--cell-min-width--base);
+  --pf-c-table__sticky-column--Left: 0;
+  --pf-c-table__sticky-column--ZIndex: var(--pf-global--ZIndex--sm);
```
[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/Table/table-scrollable.scss:5]: Doesn't look like the `ZIndex` value makes a difference. Dropdown support is going to require an enhancement or use of a popover.

File: src/patternfly/components/Table/table-scrollable.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      }
+    }
+  }
+}
+
+.pf-c-scroll-outer-wrapper {
+  display: flex;
+  flex-direction: column;
+  max-width: 100%;
+  min-height: 400px;
```
[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/Table/table-scrollable.scss:47]: I think these, at some point, will be generally usable as many components could benefit from becoming scrollable at content overflow.

[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/Table/table-scrollable.scss:47]: Although, in the future, we could convert this to its own component... WDYT @mcoker

File: src/patternfly/components/Table/templates/table--scrollable--tbody--tr.hbs (hunk 2 above)
Comment (mattnolting) [reply src/patternfly/components/Table/templates/table--scrollable--tbody--tr.hbs:8]: Updated this to illustrate `MinWidth` and `Left` relationship

**Review by mcoker — APPROVED** [review]
L🍕TM! Great work!!!

**Pull request merged**